snapshot/driver.go

```diff
@@ -0,0 +1,310 @@
+type Driver interface {
+	// Prepare returns a set of mounts corresponding to an active snapshot
+	// transaction, identifieid by the provided transaction key.
+	//
+	// If a parent is provided, after performing the mounts, the destination
+	// will start with the content of the parent. Changes to the mounted
+	// destination will be captured in relation to the provided parent. The
+	// default parent, "", is an empty directory.
+	//
+	// The changes may be saved to a new snapshot by calling Commit. When one
+	// is done with the transaction, Remove should be called on the key.
+	//
+	// Multiple calls to Prepare or View with the same key should fail.
+	Prepare(key, parent string) ([]containerd.Mount, error)
+
+	// View behaves identically to Prepare except the result may not be committed
+	// back to the snapshot manager. View returns a readonly view on the
+	// parent, with the transaction tracked by the given key.
+	//
+	// This method operates identically to Prepare, except that Mounts returned
+	// may have the readonly flag set. Any modifications to the underlying
+	// filesystem will be ignored.
+	//
+	// Commit may not be called on the provided key. To collect the resources
+	// associated with key, Remove must be called with key as the argument.
+	View(key, parent string) ([]containerd.Mount, error)
+
+	// Commit captures the changes between key and its parent into a snapshot
+	// identified by name.  The name can then be used with the driver's other
+	// methods to create subsequent snapshots.
+	//
+	// A snapshot will be created under name with the parent that started the
+	// transaction.
+	//
+	// Commit may be called multiple times on the same key. Snapshots created
+	// in this manner will all reference the parent used to start the
+	// transaction.
+	Commit(name, key string) error
+
+	// Mounts returns the mounts for the transaction identified by key. Can be
+	// called on an read-write or readonly transaction.
+	//
+	// This can be used to recover mounts after calling View or Prepare.
+	Mounts(key string) ([]containerd.Mount, error)
+
+	// Remove abandons the transaction identified by key. All resources
+	// associated with the key will be removed.
+	Remove(key string) error
+
+	// Parent returns the parent of snapshot identified by name.
+	Parent(name string) (string, error)
+
+	// Exists returns true if the snapshot with name exists.
+	Exists(name string) bool
+
+	// Delete the snapshot idenfitied by name.
+	//
+	// If name has children, the operation will fail.
+	Delete(name string) error
+
+	// TODO(stevvooe): The methods below are still in flux. We'll need to work
+	// out the roles of active and committed snapshots for this to become more
+	// clear.
+
+	// Walk the committed snapshots.
+	Walk(fn func(name string) error) error
+
+	// Active will call fn for each active transaction.
+	Active(fn func(key string) error) error
+}
+
+// DriverSuite runs a test suite on the driver given a factory function.
```

Comment:
Because you need to be able to import it into other packages and transitively built packages may not have build tags with it. I'll play around with it to make sure.